gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/N.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.process.traversal;
+
+import org.apache.tinkerpop.gremlin.process.traversal.step.map.AsNumberStep;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * Tokens that are used to denote different units of number.
+ * Used with {@link AsNumberStep} step.
+ */
+public enum N {
```

Comment:
Radical idea at this point but does `N` need to exist at all? It really doesn't serve any functional purpose that simply referencing the `Number` class would not already do. would it be more clean to only define "N" where necessary? Like, the Grammar would need an `N` to understand `int`, `float`, etc. and Javascript would need a way to reference `BigDecimal` or whatever, but does Java (or .NET) itself need a redefinition of number types?